src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCrusher.java

```diff
@@ -94,25 +95,23 @@ public BuildingCrusher(final IColony c, final BlockPos l)
      */
     private void loadCrusherMode()
     {
+        this.recipes.clear();
+        checkForWorkerSpecificRecipes();
```

Comment:
does this here load the recipes into "this.recipes" ?

Reply:
This concerns me, as this.recipes is a persisted collection, with a lot of behavioral semantics around it. While I think it would work for now, I'd rather see us just use what is in this.recipes.  What is the bad behavior this clear is fixing?

Reply:
This is cleared because datapacks can be reloaded for the crusher I believe

Reply:
We already handle datapack changes in the recipe system. checkForWorkerSpecificRecipes does full maintenance, and runs on colony tick.

Reply:
Yes, it reloads this.recipes.  I'm not entirely sure why, but without this specific reload I found hundreds of repeats of the same recipes stored in there (as well as garbage recipes that returned null after trying to resolve the token).  This fixes all of that.

It does have the downside of generating an essentially random order of the recipes and not remembering any order that the player has set, but at least as long as the crafting is mode-based this doesn't matter.